classes/order/Order.php

```diff
@@ -1097,6 +1098,36 @@ public static function getOrderByCartId($id_cart)
     }
 
     /**
+     * Get an order object by its cart id
+     *
+     * @param int $id_cart Cart id
+     * @return OrderCore
+     */
+    public static function getByCartId($id_cart)
+    {
+        $id_order = self::getIdByCartId($id_cart);
+
+        return new self($id_order);
+    }
+
+    /**
+     * Get the order id by its cart id
+     *
+     * @param int $id_cart Cart id
+     * @return int $id_order
+     */
+    public static function getIdByCartId($id_cart)
+    {
+        $sql = 'SELECT `id_order`
+				FROM `'._DB_PREFIX_.'orders`
+				WHERE `id_cart` = '.(int)$id_cart
+            .Shop::addSqlRestriction();
+        $result = Db::getInstance()->getRow($sql);
```

Comment:
Hmm after rethinking it maybe we shouldn't change. If you do, you'll get `0` instead of `false` if nothing is found. I'm not sure we want to change this behavior.